app/src/main/java/ru/vssemikoz/newsfeed/data/NewsApiRepository.java

```diff
@@ -5,45 +5,48 @@
 import retrofit2.Call;
 import retrofit2.Callback;
 import retrofit2.Response;
-import ru.vssemikoz.newsfeed.MainApplication;
+import ru.vssemikoz.newsfeed.api.NewsApi;
 import ru.vssemikoz.newsfeed.models.Category;
 import ru.vssemikoz.newsfeed.models.NewsApiResponse;
 
-public class NewsApiRepository {
-    private MainApplication mainApplication;
+public class NewsApiRepository implements ApiRepository {
+    private NewsApi newsApi;
+    private String apiKey;
+    private String countryKey;
 
-    public interface RequestListener {
-        void onApiRequestSuccess(Response<NewsApiResponse> response);
-
-        void onApiRequestFailure(Throwable t);
-    }
-
-    public NewsApiRepository(MainApplication mainApplication) {
-        this.mainApplication = mainApplication;
+    // TODO: 26.03.2020 remove keys from constructor
```

Comment:
You should remove not only keys from here but provide `NewsApi` via Dagger as well.